CTS1.rst

```diff
@@ -0,0 +1,564 @@
+
+The parameter is first a byte indicating whether to remove or add an operator, where if the byte is 0 the sender is removing an operator, if the byte is 1 the sender is adding an operator.
+The followed is the operator address :ref:`CTS-Address` to add or remove as operator for the sender.
+
+Requirements
+~~~~~~~~~~~~
+
+- The contract function MUST reject if the sender address is the same as the operator address with error :ref:`OPERATOR_IS_SENDER<CTS-rejection-errors>`.
+
+.. note::
+
+  Operators are not set per token ID, and an operator can control any token type of the owner address.
+  This was chosen to require less on the contract implementation and also simplify off-chain integration.
+  If needed a more fine grained authentication system can still exist next to the operators.
+
+.. _CTS-functions-balanceOf:
+
+``balanceOf``
+^^^^^^^^^^^^^
+
+Query balances of a list of addresses and token IDs, the result is then send back the sender.
+
+Parameter
+~~~~~~~~~
+
+The parameter consists of a name of the receive function to callback with the result and a list of token ID and address pairs.
+It is serialized as: :ref:`CTS-ReceiveFunctionName` followed by 1 byte for the number of queries and then this number of queries.
+A queries is serialized as :ref:`CTS-TokenID` followed by :ref:`CTS-Address`.
+
+.. note::
+
+  Be aware of the size limit on contract function parameters which currently is 1024 bytes, depending on the byte size of the Token ID and the name of the receive function.
+
+Callback parameter
+~~~~~~~~~~~~~~~~~~
+
+The parameter for the callback receive function is a list of query and token amount pairs.
+It is serialized as: 1 byte for the number of query-amount pairs and then this number of pairs.
+A query-amount pair is serialized as a :ref:`CTS-TokenID`, an address :ref:`CTS-Address` and a :ref:`CTS-TokenAmount`.
+
+Requirements
+~~~~~~~~~~~~
+
+- The contract function MUST reject if the sender is not a contract address with error :ref:`CONTRACT_ONLY<CTS-rejection-errors>`.
+- The contract function MUST reject if any of the queries fail.
+- A query MUST fail if the token ID is unknown with error: :ref:`INVALID_TOKEN_ID<CTS-rejection-errors>`.
+
+Logged events
+-------------
+
+The idea of the logged events for this specification is for off-chain applications to be able to track balances and operators without knowledge of the contract specific implementation details.
+For this reason it is important to log events in any custom functionality for the token contract, if these modifies balances or operators.
+
+It MUST be safe for off-chain applications to assume a contract implementing this specification and no events logged have zero tokens and no operators enabled for any address.
+
+.. Other events custom to the contract implementation MUST be safe for the off-application to ignore.
```

Comment:
This is a RST comment ment for myself, and it should have been removed.

I'll put a requirement for accumulating the balance from the CTS events only, instead of something like this.